### PR TITLE
Implement overdue unit tests on API component and refactor

### LINKED ISSUE #339: Implement overdue unit tests on API component
**Describe the issue**
As a dev, I'd like to add the missing / overdue unit tests for the existing functionalities on API component, as  it is a necessary condition to integrating new features effectively at the core values of test-driven development.

**Additional context**
...

**Definition of done**
When the missing / overdue unit tests are added.

### PR DESCRIPTION
**Pull request by yhfreeman**
- Added many missing unit tests for functions / methods inside api controllers and services.
- Refactored around the edit request business logics in order to write proper unit tests for them.
  - Unified 2 endpoints in 1 and modify the frontend accordingly.
    - project-profile provisioner-related update, project-profile non provisioner-related update >> project-profile update
  - Rename some endpoints for consistency and modify the frontend accordingly.
  - End result:
```
// may involve provisioner-related changes
router.put('/:profileId', authorize(), asyncMiddleware(updateProjectProfile));

// may involve provisioner-related changes
router.post('/:profileId/contacts', authorize(), asyncMiddleware(updateProfileContacts));

// will invovle provisioner-related changes
router.post('/:profileId/quota-size', authorize(), asyncMiddleware(updateProfileQuotaSize));
```
- Disabled the ability to update the project-profile name from Registry API and added unit test.

- Some linter changes on the frontend resulted from running
  - `npm run lint -- --fix`
  - `npm run format -- -write`

Fixes #339 

Due to the pruning nature of unit tests
/bot-ignore-length

### CODE REVIEW

File: api/__tests__/fixtures/get-jwt-payload-user.json [hunk 1]
```diff
@@ -27,7 +22,7 @@
     "email_verified": true,
     "name": "Jane Doe",
     "preferred_username": "jane1100@idir",
-    "given_name": "Yuhan",
-    "family_name": "Freeman",
+    "given_name": "Jane",
+    "family_name": "Doe",
     "email": "jane@example.com"
 }
```
Comment (sbathgate) [inline api/__tests__/fixtures/get-jwt-payload-user.json:28]:
```suggestion
}

```
Nit: ending newline

File: api/__tests__/fixtures/insert-profile.json [hunk 2]
```diff
@@ -20,5 +20,6 @@
   "idmKeyCloak": false,
   "idmActiveDir": false,
   "other": "This is a test",
-  "primaryClusterName": "silver"
+  "primaryClusterName": "silver",
+  "migratingLicenseplate": "abc123"
 }
```
Comment (sbathgate) [inline api/__tests__/fixtures/insert-profile.json:25]:
```suggestion
}

```
Nit: ending newline

**Review by sbathgate — APPROVED** [review]
Overall this looks great. As discussed a couple of small nit picks, and a consideration on how to remove Lodash requirements from utils. Nothing that would block this though.

File: api/__tests__/fixtures/get-jwt-payload-user.json (hunk 1 above)
Comment (yhfreeman) [reply api/__tests__/fixtures/get-jwt-payload-user.json:28]: Looks like the rule of ending newlines doesn't go on .json file easily, at least not within the VS code?

File: api/src/libs/utils.ts [hunk 3]
```diff
@@ -13,39 +13,27 @@
...
-  provider: string,
-  email: string,
-}
+import { difference } from 'lodash';
 
-interface NatsContactObject {
-  contact: NatsContactDetails[];
-}
+export const validateObjProps = (requiredFields: string[], pojo: object): Error | undefined => {
+  const diff = difference(requiredFields, Object.keys(pojo));
```
[the remark this answers is not in the document]
Comment (yhfreeman) [reply api/src/libs/utils.ts:21]: There's currently another spot where lodash is being used to switch camelCase. We could perhaps open a separate PR if/when we are removing lodash dependency?

**Comment by yhfreeman** [comment]
@sbathgate Thank you very much for your review!
I tested this with the gitops-lab repo, and realized the nats message contact edit fields aren't quite right.
Will request another review once a fix is provided.

**Comment by yhfreeman** [comment]
@sbathgate as per discussed in RC, the above-mentioned bug magnify some bigger issues, which may be blockers to further implementing the multiple technical leads and multi-cluster support.
The proposed set of changes are to address such issues:
- Modularize the way bot structure gets generated for create, edit and sync.
  - Removing using nats_context in Request as it should only be used for auditing.
- Remove the original singular contact update, as it no longer works with the gitops patterns (potential race conditions).
- Centralize the JSON.stringify and parse inside Request model.
- Refactor namespace-set service methods -> Namespace model methods.
- Refactor update contacts so it will work with a list of contacts.

Thanks.

**Review by sbathgate — APPROVED** [review]
Overall, this is obviously an immense amount of changes so difficult to fully vette, but throughout my local development work I haven't recognised any glaring issues so this should all be good to merge.

**Pull request merged**